src/api/facade.js

```diff
@@ -57,64 +47,52 @@ export const checkValidHash = (data: string) => {
 
 export const checkAndFacadeTransactionAsync = async (
   tx: RawTransaction,
+  bestBlockNum: number,
 ): Promise<Transaction> => {
-  assert.assert(
-    tx.inputs_address.length === tx.inputs_amount.length,
-    'Invalid data from server (inputs mismatch)',
-    tx.inputs_address,
-    tx.inputs_amount,
-  )
-  assert.assert(
-    tx.outputs_address.length === tx.outputs_amount.length,
-    'Invalid data from server (outputs mismatch)',
-    tx.outputs_address,
-    tx.outputs_amount,
-  )
-
-  assert.assert(
-    checkNonNegativeInt(tx.best_block_num),
-    'Invalid best_block_num',
-    tx.best_block_num,
-  )
-  const bestBlockNum = parseInt(tx.best_block_num, 10)
-
-  assert.assert(
-    tx.tx_state !== 'Successful' || checkNonNegativeInt(tx.block_num),
-    'Invalid block_num',
-    tx.block_num,
-  )
-
-  const blockNum = tx.block_num ? parseInt(tx.block_num, 10) : null
-
-  tx.inputs_amount.forEach((amount) => {
-    assert.assert(checkNonNegativeInt(amount), 'Invalid input amount', amount)
+  // const blockNum = tx.block_num != null ? parseInt(tx.block_num, 10) : null
+  // const txOrdinal = tx.tx_ordinal != null ? parseInt(tx.tx_ordinal, 10) : null
+
+  tx.inputs.forEach((i) => {
+    assert.assert(
+      checkNonNegativeInt(i.amount),
+      'Invalid input amount',
+      i.amount,
+    )
   })
 
-  tx.outputs_amount.forEach((amount) => {
-    assert.assert(checkNonNegativeInt(amount), 'Invalid output amount', amount)
+  tx.outputs.forEach((o) => {
+    assert.assert(
+      checkNonNegativeInt(o.amount),
+      'Invalid output amount',
+      o.amount,
+    )
   })
 
   await Promise.all(
-    tx.inputs_address.map(async (address) => {
+    tx.inputs.map(async (input) => {
       assert.assert(
-        await isValidAddress(address),
+        await isValidAddress(input.address),
         'Invalid input address',
-        address,
+        input.address,
       )
     }),
   )
 
   await Promise.all(
-    tx.outputs_address.map(async (address) => {
+    tx.outputs.map(async (output) => {
       assert.assert(
-        await isValidAddress(address),
+        await isValidAddress(output.address),
         'Invalid output address',
-        address,
+        output.address,
       )
     }),
   )
 
-  assert.assert(checkISO8601Date(tx.time), 'Invalid time', tx.time)
+  assert.assert(
+    tx.tx_state !== 'Successful' || checkISO8601Date(tx.time),
```

Comment:
Maybe add a comment saying this field is only present if the tx is successful which is why we need this `||`?